src/circuit/gadgets/constraints.rs

```diff
@@ -1,5 +1,10 @@
 // Initially taken from: rust-fil-proofs/storage-proofs-core/src/gadgets/
 
+use crate::circuit::gadgets::data::GlobalAllocations;
+use crate::circuit::gadgets::pointer::AllocatedPtr;
```

Comment:
As mentioned throughout, I think this import is evidence that the code using it is more complicated than necessary. The functions in this module should be 'generic' and not Lurk-specific.

Reply:
fixed